src/python/pants/base/exceptions.py

```diff
@@ -3,7 +3,12 @@
 
 from __future__ import annotations
 
-from typing import Iterable
+from typing import TYPE_CHECKING, Iterable
+
+from pants.util.strutil import bullet_list, softwrap
+
+if TYPE_CHECKING:
+    from pants.build_graph.address import Address
```

Comment:
While this will work fine, it will still end up forming a cycle between `src/python/pants/base` and `src/python/pants/build_graph` ... might be better to move or use a new exception.